sdk/canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/domain/domain.scala

```diff
@@ -142,12 +146,37 @@ final case class DisclosedContract(
     driverMetadata: Bytes,
     transactionVersion: TransactionVersion,
 ) {
-  def toLf: LfDisclosedContract =
-    LfDisclosedContract(
-      templateId,
-      contractId,
-      argument,
-      keyHash,
+
+  private[this] def keyWithWithMaintainer =
+    (keyValue, keyMaintainers) match {
+      case (None, None) => None
+      case (Some(value), Some(maintainers)) =>
+        Some(GlobalKeyWithMaintainers.assertBuild(
+          templateId,
+          value,
+          maintainers,
+          packageName
+        )
+        )
+      case _ =>
+        throw new Error("unexpected mismatch between keyValue and keyMaintainers")
```

Comment:
I'd add the received args in the error for debugging purposes

Reply:
This will be overwritten by the next code drop, so I am quite to let it like that.
I have a [companion](https://github.com/DACH-NY/canton/compare/remy-disclosure?expand=1) branch for canton where this disapear anyway.